### PR TITLE
ArC: update annotation transformation documentation

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Review by mkouba — APPROVED** [review]
Thanks!

**Review by manovotn — APPROVED** [review]

File: docs/src/main/asciidoc/cdi-integration.adoc [hunk 1]
```diff
@@ -169,20 +169,20 @@ With an `AnnotationsTransformerBuildItem` it's possible to override the annotati
...
 [source,java]
 ----
 @BuildStep
 AnnotationsTransformerBuildItem transform() {
-   return new AnnotationsTransformerBuildItem(AnnotationsTransformer.appliedToClass() <1>
-        .whenClass(c -> c.name().toString().equals("org.acme.Bar")) <2>
-        .thenTransform(t -> t.add(MyInterceptorBinding.class))); <3>
+    return new AnnotationsTransformerBuildItem(AnnotationTransformation.forClasses() // <1>
+        .whenClass(DotName.createSimple("org.acme.Bar")) // <2>
+        .transform(t -> t.add(MyInterceptorBinding.class))); // <3>
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-integration.adoc:181]: For the future, IIRC, you don't need the `//`. The callouts are automatically removed when you copy/paste the code.
It doesn't hurt though.

Comment (Ladicek) [reply docs/src/main/asciidoc/cdi-integration.adoc:181]: I know they are not necessary, but they help IDEA understand the code. That's why I've been using them everywhere :-)

Comment (gsmet) [reply docs/src/main/asciidoc/cdi-integration.adoc:181]: OK, makes sense then!

**Pull request merged**